snapcraft_legacy/storeapi/v2/validation_sets.py

```diff
@@ -14,161 +14,135 @@
-            "type": self.assertion_type,
-            "series": self.series,
-        }
-
-    def __eq__(self, other) -> bool:
-        if not isinstance(other, BuildAssertion):
-            return False
-
-        return (
-            self.name == other.name
-            and self.account_id == other.account_id
-            and self.authority_id == other.authority_id
-            and self.revision == other.revision
-            and self.sequence == other.sequence
-            and self.snaps == other.snaps
-            and self.timestamp == other.timestamp
-            and self.assertion_type == other.assertion_type
-            and self.series == other.series
-        )
-
-    def __repr__(self) -> str:
-        return f"<{self.__class__.__name__}: {self.name!r}>"
-
-    def __init__(
-        self,
-        *,
-        account_id: str,
-        authority_id: str,
-        name: str,
-        revision: str,
-        sequence: str,
-        snaps: List[Snap],
-        timestamp: str,
-        assertion_type: str,
-        series: str = "16",
-    ):
-        self.account_id = account_id
-        self.authority_id = authority_id
-        self.name = name
-        self.revision = revision
-        self.sequence = sequence
-        self.snaps = snaps
-        self.series = series
-        self.timestamp = timestamp
-        self.assertion_type = assertion_type
-
-
-class ValidationSets:
-    @classmethod
-    def unmarshal(cls, payload: Dict[str, Any]) -> "ValidationSets":
-        return cls(
-            assertions=[
-                BuildAssertion.unmarshal(a["headers"])
-                for a in payload.get("assertions", list())
-            ]
-        )
-
-    def marshal(self) -> Dict[str, Any]:
-        return {"assertions": [{"headers": a.marshal()} for a in self.assertions]}
-
-    def __repr__(self) -> str:
-        assertion_count = len(self.assertions)
-        return f"<{self.__class__.__name__}: assertions {assertion_count!r}>"
-
-    def __init__(self, assertions: List[BuildAssertion]) -> None:
-        self.assertions = assertions
+    name: SnapName
+    """Snap name"""
+
+    id: SnapId | None
+    """Snap ID"""
+
+    presence: Literal["required", "optional", "invalid"] | None
+    """Snap presence"""
+
+    revision: int | None
+    """Snap revision"""
+
+
+class EditableBuildAssertion(models.CraftBaseModel):
+    """Subset of a build assertion that can be edited by the user.
+
+    https://dashboard.snapcraft.io/docs/reference/v2/en/validation-sets.html#request-json-schema
+    """
+    account_id: str
+    """The "account-id" assertion header"""
+
+    name: str
+    """The "name" assertion header"""
+
+    revision: str | None
+    """The "revision" assertion header"""
+
+    sequence: int
+    """The "sequence" assertion header"""
+
+    snaps: Annotated[list[Snap], pydantic.Field(min_items=1)]
+    """List of snaps in a Validation Set assertion"""
+
+    def marshal_as_str(self) -> Dict[str, Any]:
```

Comment:
Ah I see what this is doing. Maybe we should be a bit more explicit about the dict values being converted to strings in the function name?